lib/Dialect/Poly/IR/Polynomial.cpp

```diff
@@ -66,6 +67,39 @@ void Polynomial::print(raw_ostream &os) const {
   }
 }
 
+std::string Polynomial::toIdentifier() const {
```

Comment:
just for simplicity could you reuse the print logic with a private hidden printer that takes the os and the separator? ` + ` or `_`?

It looks like you have different ways of handling the exponent multiplicatin though (`**`) is that for simplicity or because of some reason not to have asteriks?

Reply:
Done.